### PR TITLE
[meshcop] include KEK TLV for only JOIN_FIN.rsp message

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR fixes the issues that the `KEK TLV` is not included for only `JOIN_FIN.rsp` messages.

### CODE REVIEW

File: src/library/commissioning_session.cpp [hunk 1]
```diff
@@ -227,12 +200,79 @@ Error CommissioningSession::SendJoinFinResponse(const coap::Request &aJoinFinReq
...
+    , mPeerAddr(aPeerAddr)
+    , mPeerPort(aPeerPort)
+    , mLocalAddr(aLocalAddr)
+    , mLocalPort(aLocalPort)
+{
+    mIsConnected = true;
+
+    int fail = event_assign(&mEvent, mEventBase, -1, EV_PERSIST | EV_READ | EV_WRITE | EV_ET, HandleEvent, this);
+    ASSERT(fail == 0);
+    ASSERT((fail = event_add(&mEvent, nullptr)) == 0);
```
Comment (jwhui) [inline src/library/commissioning_session.cpp:226]: Avoid including logic in assert statement?

File: src/library/socket_test.cpp [hunk 2]
```diff
@@ -44,6 +48,109 @@ static uint16_t    kServerPort = 9527;
...
+{
+    ASSERT(!aBuf.empty());
+    return Send(aBuf.data(), aBuf.size());
+}
+
+int MockSocket::Receive(ByteArray &aBuf)
+{
+    uint8_t buf[512];
+    int     rval;
+    while ((rval = Receive(buf, sizeof(buf))) > 0)
```
Comment (jwhui) [inline src/library/socket_test.cpp:129]: nit: add empty line after variable declaration block

File: src/common/time.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace ot {
+
+namespace commissioner {
+
+std::string TimePointToString(const TimePoint &aTimePoint)
+{
+    struct tm   localTime;
+    std::time_t time = Clock::to_time_t(aTimePoint);
+    localtime_r(&time, &localTime);
```
Comment (jwhui) [inline src/common/time.cpp:49]: nit: add empty line after variable declaration block

File: src/library/commissioning_session.cpp [hunk 4]
```diff
@@ -135,16 +104,20 @@ Error CommissioningSession::SendRlyTx(const ByteArray &aBuf)
...
+        VerifyOrExit(!mDtlsSession->GetKek().empty(), error = Error::kInvalidState);
+
+        LOG_DEBUG("include KEK TLV in RLY_TX.ntf message");
         SuccessOrExit(error = AppendTlv(rlyTx, {tlv::Type::kJoinerRouterKEK, mDtlsSession->GetKek()}));
     }
 
     mCommImpl.mBrClient.SendRequest(rlyTx, nullptr);
 
-    LOG_DEBUG("sent RLY_TX.ntf: joinerIID={}, length={}", utils::Hex(GetJoinerIid()), aBuf.size());
+    LOG_DEBUG("sent RLY_TX.ntf: CommissioningSessionState={}, joinerIID={}, length={}", mDtlsSession->GetStateString(),
```
Comment (jwhui) [inline src/library/commissioning_session.cpp:119]: Combine `aIncludeKek` log output from above as a field in this one?

File: src/library/commissioning_session.cpp [hunk 5]
```diff
@@ -227,12 +200,79 @@ Error CommissioningSession::SendJoinFinResponse(const coap::Request &aJoinFinReq
...
+    : Socket(aCommissioningSession.mCommImpl.GetEventBase())
+    , mCommissioningSession(aCommissioningSession)
+    , mPeerAddr(aPeerAddr)
+    , mPeerPort(aPeerPort)
+    , mLocalAddr(aLocalAddr)
+    , mLocalPort(aLocalPort)
+{
+    mIsConnected = true;
+
+    int fail = event_assign(&mEvent, mEventBase, -1, EV_PERSIST | EV_READ | EV_WRITE | EV_ET, HandleEvent, this);
```
Comment (jwhui) [inline src/library/commissioning_session.cpp:224]: nit: variable declaration at top of scope?

File: src/common/time.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * @file
+ *   This file implements time measurements.
+ */
+
+#include "time.hpp"
+
+#include <time.h>
+
+#include <iomanip>
```
Comment (jwhui) [inline src/common/time.cpp:38]: nit: c++ headers before c headers?

File: src/library/commissioning_session.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/library/commissioning_session.cpp:226]: The `ASSERT` here is implemented as `abort()` if the condition is not satisfied (it allows side-effect). The name may be not suitable, I have been thinking rename it to `Check(...)` or `VerifyOrDie(...)`. What do you think?

File: src/common/time.cpp (hunk 6 above)
Comment (wgtdkp) [reply src/common/time.cpp:38]: okay, how about apply this change to all source files in a separate PR?

File: src/library/commissioning_session.cpp (hunk 1 above)
Comment (jwhui) [reply src/library/commissioning_session.cpp:226]: Of the options, I think `VerifyOrDie()` is more descriptive and better than `ASSERT`.

File: src/common/time.cpp (hunk 6 above)
Comment (jwhui) [reply src/common/time.cpp:38]: I'm fine to fix this globally in a separate PR.

File: src/library/commissioning_session.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/library/commissioning_session.cpp:226]: okay, I will do renaming it in separate PR.

**Pull request merged**